src/layer-composer/generators/vessel-events/vessel-events.ts

```diff
@@ -0,0 +1,132 @@
+import { GeneratorConfig } from 'layer-composer/types'
+import { FeatureCollection } from 'geojson'
+import { GeoJSONSourceRaw } from 'mapbox-gl'
+import { Dictionary } from 'types'
+
+export const VESSEL_EVENTS_TYPE = 'VESSEL_EVENTS'
+
+export interface VesselEventsGeneratorConfig extends GeneratorConfig {
+  data: FeatureCollection
+}
+
+const BASEMAP_COLOR = '#00265c'
+
+class VesselsEventsGenerator {
+  type = VESSEL_EVENTS_TYPE
+
+  _getStyleSources = (layer: VesselEventsGeneratorConfig) => {
+    const { id, data } = layer
+
+    if (!data) {
+      console.warn(`${VESSEL_EVENTS_TYPE} source generator needs geojson data`, layer)
+      return []
+    }
+
+    const source: GeoJSONSourceRaw = {
+      type: 'geojson',
+      data: data || null,
+    }
+    return [{ id, ...source }]
+  }
+
+  _getStyleLayers = (layer: VesselEventsGeneratorConfig) => {
+    if (!layer.data) {
+      console.warn(`${VESSEL_EVENTS_TYPE} source generator needs geojson data`, layer)
+      return []
+    }
+
+    const activeFilter = ['case', ['==', ['get', 'active'], true]]
+    const layers: any[] = [
+      {
+        id: 'vessel_events_bg',
+        type: 'circle',
+        source: layer.id,
+        paint: {
+          'circle-color': ['get', 'color'],
+          'circle-stroke-width': 2,
+          'circle-stroke-color': [...activeFilter, 'rgba(0, 193, 231, 1)', BASEMAP_COLOR],
+          'circle-radius': [...activeFilter, 12, 5],
+        },
+      },
+      {
+        id: 'vessel_events',
+        source: layer.id,
+        type: 'symbol',
+        layout: {
+          'icon-allow-overlap': true,
+          'icon-image': ['get', 'icon'],
+          'icon-size': [...activeFilter, 1, 0],
+        },
+      },
+    ]
+    return layers
+  }
+
+  getStyle = (layer: VesselEventsGeneratorConfig) => {
+    return {
+      id: layer.id,
+      sources: this._getStyleSources(layer),
+      layers: this._getStyleLayers(layer),
+    }
+  }
+}
+
+export default VesselsEventsGenerator
+
+type AuthorizationOptions = 'authorized' | 'partially' | 'unmatched'
+type Event = {
+  type: string
+  active: boolean
+  timestamp: number
+  authorizationStatus: AuthorizationOptions
+  coordinates: {
+    lng: number
+    lat: number
+  }
+}
+
+const EVENTS_COLORS: Dictionary<string> = {
```

Comment:
Not do do it now, but this configuration could be extended from the generator data config to be more flexible in the future.

Reply:
I don't think it should. It's not a generic vessel events layer, it's supposed IMHO to be a GFW vessel events layer, so using brand colors. (Same way it's done in the basemap layers)